### PR TITLE
Ticket 890: use new orientation definition for asymmetric shapes

### PR DESCRIPTION
**Pull request by pkienzle**
Update asymmetric models to use the new definition of orientation.   First rotate the shape about the z axis by angle psi.  Then rotate the shape about the y axis by angle theta.  Then rotate the shape about the z axis again by angle phi.

Note that rotation about z should rotate the pattern in the detector plane.  This means that:

(1) for theta=0, both phi and psi should act to equivalently to rotate the image on the detector.  
(2) phi should rotate the image on the detector regardless of theta and phi.

These two points were checked for all models.

The rectangular prism is particularly easy to check.  With combinations of theta, phi and psi equal to 0 or 90 you should get a simple pattern, with the repeat spacing inversely proportional to 1/a, 1/b and 1/c for whichever pair ab, ac or bc is parallel to the detector.  Similarly for other models, but the patterns are more complicated.

For elliptical models check that the equivalent circular model matches, both in 1D and in 2D.  I found a number of bugs this way.  In particular, the 1D triaxial ellipse model was not giving the correct result.  Try for all radii=57 and it should now match the result for a sphere.

Some doc strings were updated, particularly for triaxial ellipsoid, which uses new equations.  Others still need work.  In particular, the orientation diagrams have not been updated with the new angle definitions.

Re: triaxial ellipsoid.  You should be able to set polar radius and the equat_major radius equal, rotate by 90-theta and get the same pattern as the ellipsoid with polar radius set to equat_minor and equatorial radius set to equat_major.  This seems to work with 2D patterns.  Since the 1D pattern integrates over all possible orientations it doesn't matter which order the three axes a, b, and c are assigned for parallelepiped and triaxial ellipsoid models.  The same 1-D pattern is computed regardless.

### CODE REVIEW

**Comment by pkienzle** [comment]
2D parallelepiped and core shell parallelepiped are failing because the definition of orientation has changed.  Need to update theta, phi,psi in the test so that the new values correspond to the same orientation as the old values.  Technically we ought to update convert.py as well, but the chances are small that anyone has saved 2D parallelepiped projects that they need to reload.

File: sasmodels/models/triaxial_ellipsoid.py [hunk 1]
```diff
@@ -68,16 +107,24 @@
...
+from numpy import inf, sin, cos, pi
 
 name = "triaxial_ellipsoid"
 title = "Ellipsoid of uniform scattering length density with three independent axes."
 
-description = """\
+description = """
 Note: During fitting ensure that the inequality ra<rb<rc is not
 	violated. Otherwise the calculation will
 	not be correct.
```
Comment (pkienzle) [inline sasmodels/models/triaxial_ellipsoid.py]: Description isn't correct anymore

File: sasmodels/models/core_shell_bicelle_elliptical.c [hunk 2]
```diff
@@ -73,17 +51,17 @@ double
...
         si2 = sas_sinx_x(sinarg2);
         for(int j=0;j<76;j++) {
             //76 gauss points for the inner integral (WAS 20 points,so this may make unecessarily slow, but playing safe)
             //const double beta = ( Gauss76Z[j]*(vbj-vaj) + vaj + vbj )/2.0;
             const double beta = ( Gauss76Z[j] +1.0)*M_PI_2;
             const double rr = sqrt(rA - rB*cos(beta));
-            double besarg1 = qq*rr*sin_alpha;
-            double besarg2 = qq*(rr+radthick)*sin_alpha;
+            double besarg1 = q*rr*sin_alpha;
+            double besarg2 = q*(rr+thick_rim)*sin_alpha;
```
Comment (pkienzle) [inline sasmodels/models/core_shell_bicelle_elliptical.c]: Check against 2D model.  I suspect that rr+thick_rim is not correct.  Also, check if it should be:
     rr = sqrt(rA - rB*square(cos(beta)))

Comment (RichardHeenan) [inline sasmodels/models/core_shell_bicelle_elliptical.c]: I think the thick_rim part is very likely OK. Though the core is elliptical, the rim is constant thickness, thus if we make the whole bicelle the same sld, the comparison with elliptical_cylinder or core_shell_bicelle has to be done with a different axial ratio to compensate. 
Need to check whether the 2d version that you sorted for me does actually do this.

If the cos(beta) bit is wrong then it is also wrong in elliptical cylinder.

Have renamed rA to r2A to remind us that it is a length^2

**Pull request merged**